pkg/operator/controller/ingress/controller.go

```diff
@@ -473,17 +473,30 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 			}
 			switch lbType {
 			case operatorv1.AWSLoadBalancerProvider:
-				// The only provider parameter that is supported
-				// for AWS is the connection idle timeout for
-				// classic ELBs.
-				var specIdleTimeout, statusIdleTimeout metav1.Duration
-				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					specIdleTimeout = specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					statusIdleTimeout = statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if specIdleTimeout != statusIdleTimeout {
+				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
+					if statusLB == nil {
```

Comment:
Optionally, if you do need that logic for `statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil ` you do it this way (but I am unsure because your code had both `!= nil` and `== nil`:
```suggestion
if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil {
  if statusLB.ProviderParameters.AWS.Type != operatorv1.AWSNetworkLoadBalancer || statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil  {
     statusLB.ProviderParameters = specLB.ProviderParameters
     changed = true
}
```

Reply:
At minimum, consider line 458 and that `statusLB` can never be null, this is a redundant check in each `else if` statement.